### ISSUE TITLE
Empty/Unit type for (:>) combinator

### ISSUE THREAD

**Issue by mchaver** [haskell-servant/servant#863]
It might be useful to have a unit type for `(:>)` where all of the following routes are equivalent, particularly if you try to make routes with type level programming.

```haskell
"hello" :> Get '[JSON] Text
"hello" :> Empty :> Get '[JSON] Text
Empty :> "hello" :> Get '[JSON] Text
```

Are there any drawbacks or issues this would create?

**Comment by 3noch** [comment]
Having it at leaf could just make the endpoint unreachable / remove it from the tree entirely.

**Comment by alpmestan** [comment]
Would feel a little redundant with `EmptyAPI`, which is our unit for `:<|>`. Good luck finding a good name =)

**Comment by phadej** [comment]
We can repurpose `EmptyAPI`, but indeed good name would be good :roll_eyes:

**Comment by mchaver** [comment]
If it is separate from `EmptyAPI` then `EmptySub` or `EmptySubAPI` since `:>` comes from `Servant.API.Sub`. If you are able to combine the two then `Empty` might suffice, though maybe we would want a type synonym `EmptyAPI` for backwards compatibility?

**Comment by alpmestan** [comment]
Maybe it already works?

``` haskell
$ nix-shell -p 'haskell.packages.ghc802.ghcWithPackages (ps: with ps; [ constraints servant servant-server wai warp ])' --run 'ghci'
GHCi, version 8.0.2: http://www.haskell.org/ghc/  :? for help
Loaded GHCi configuration from /home/alp/.ghci
Prelude λ> :set -XTypeOperators
Prelude λ> import Servant
Prelude Servant λ> :set -XDataKinds
Prelude Servant λ> :kind! Server (Capture "foo" Int :> EmptyAPI)
Server (Capture "foo" Int :> EmptyAPI) :: *
= Int -> Tagged Handler EmptyServer
```

So it doesn't "skip" the endpoint(s), but you simply have to provide `emptyServer` wherever you need to supply a `Tagged Handler EmptyServer`.

**Comment by 3noch** [comment]
@alpmestan For that case that's probably actually *preferred*. It would be very hard to understand if the arity of your server changed so easily.

**Comment by alpmestan** [comment]
Yes indeed. However, this is not the behavior described in the original comment of this thread, so it appears that we do need another name... :disappointed:
